samples/excel/30-events/selection-changed-events.yaml

```diff
@@ -0,0 +1,160 @@
+order: 7
+id: excel-selection-changed-events
+name: Selection change events
+description: Registers handlers all the different `onSelectionChanged` events and displays how each event reports the selected addresses.
+author: OfficeDev
+host: EXCEL
+api_set:
+    ExcelApi: '1.7'
+script:
+    content: |
+        $("#setup").click(() => tryCatch(setup));
+        $("#register-event-handlers").click(() => tryCatch(registerEventHandlers));
+
+        async function registerEventHandlers() {
+          await Excel.run(async (context) => {
+            // Add a selection changed event handler for the binding.
+            let binding = context.workbook.bindings.getItemAt(0);
+            binding.onSelectionChanged.add(onBindingSelectionChange);
+
+            // Add a selection changed event handler for the table.
+            let table = context.workbook.tables.getItemAt(0);
+            table.onSelectionChanged.add(onTableSelectionChange);
+
+            // Add a selection changed event handler for the worksheet.
+            let sheet = context.workbook.worksheets.getItem("Sample");
+            sheet.onSelectionChanged.add(onWorksheetSelectionChange);
+
+            // Add a selection changed event handler for the worksheet collection.
+            context.workbook.worksheets.onSelectionChanged.add(onWorksheetCollectionSelectionChange);
+
+            await context.sync();
+          });
+        }
+
+        async function onBindingSelectionChange(args: Excel.BindingSelectionChangedEventArgs) {
+          await Excel.run(async (context) => {
```

Comment:
You've probably already done this, but just for sanity I'll ask, have you verified that the `async` keyword is needed here? There don't seem to be any internal calls to `await`. Same question applies to the next few handlers defined below. Then, again, maybe its a good idea to keep it anyway since in most real life `Excel.run`s it would be needed. So, this is just something to think about.

Reply:
This is definitely something to think about. I left it to maintain the typical `Excel.run` pattern we use in the TypeScript samples. Like you mentioned, most `Excel.run`s need the `async`/`await`, so this ensures there won't be any copy/paste hiccups for developers building off of this sample.

Happy to chat further and do an audit of the other samples, if needed.